### PR TITLE
Made Vector2 composite actions respect the press points of button controls used to compose the value.

### PR DESCRIPTION
**Pull request by jechter**
Fixes https://fogbugz.unity3d.com/f/cases/1131844/

### CODE REVIEW

**Review by douglas-piconi — APPROVED** [review]

File: Packages/com.unity.inputsystem/InputSystem/Actions/InputActionState.cs [hunk 1]
```diff
@@ -2089,10 +2091,14 @@ internal TValue ReadCompositePartValue<TValue>(int bindingIndex, int partNumber)
                     {
                         result = value;
                         isFirstValue = false;
+                        if (controls[controlIndex] is Controls.ButtonControl)
```
Comment (Rene-Damm) [inline Packages/com.unity.inputsystem/InputSystem/Actions/InputActionState.cs:2117]: Pattern matching obviates the need for the second cast.

`if (controls[controlIndex] is ButtonControl button)`

**Review by Rene-Damm — COMMENTED** [review]
TBH, on the fence about this.

The thing that is bothering me is that I'm more and more thinking that `pressPoint` on ButtonControl is probably a bad design and should most likely just be nuked and forgotten. I mean, how is `pressPoint` different to deadzoning we do on sticks? If it isn't, why don't we just deadzone certain buttons that are actually true axes underneath. And if we do that, there's no need to treat a button any different than any other float control here. So yeah, I'm thinking I simply went down a wrong path and introduced something that shouldn't be there in the first place.

What do you think @jechter?

**Comment by jechter** [comment]
I guess technically no, this is not different from a dead zone. Can we set up a dead zone on the control so that values are automatically processed by it without requiring further manual setup? Then that might be a more consistent approach indeed.

**Comment by Rene-Damm** [comment]
@jechter and I talked this one over quite a bit and in the end agreed to keep button press points.

**Review by Rene-Damm — APPROVED** [review]

**Comment by jechter** [comment]
FIX: Made Vector2 composite actions respect the press points of button controls used to compose the value.

**Pull request merged**